src/main/java/org/opensearch/ad/rest/handler/ModelValidationActionHandler.java

```diff
@@ -384,29 +400,36 @@ public void onResponse(SearchResponse response) {
                     listener
                         .onFailure(
                             new ADValidationException(
-                                CommonErrorMessages.MODEL_VALIDATION_FAILED_UNEXPECTEDLY,
-                                DetectorValidationIssueType.MODEL_VALIDATION_ISSUE,
+                                CommonErrorMessages.TIMEOUT_ON_INTERVAL_REC,
+                                DetectorValidationIssueType.TIMEOUT,
                                 ValidationAspect.MODEL
                             )
                         );
-                    logger.info("Timed out getting interval recommendation");
+                    logger.info(CommonErrorMessages.TIMEOUT_ON_INTERVAL_REC);
                     // keep trying higher intervals as new interval is below max, and we aren't decreasing yet
-                } else if (newInterval < MAX_INTERVAL_REC_LENGTH_IN_MINUTES && !decreasingInterval) {
-                    searchWithDifferentInterval(newInterval);
+                } else if (newIntervalMinute < MAX_INTERVAL_REC_LENGTH_IN_MINUTES && !decreasingInterval) {
+                    searchWithDifferentInterval(newIntervalMinute);
                     // The below block is executed only the first time when new interval is above max and
                     // we aren't decreasing yet, at this point
-                } else if (newInterval >= MAX_INTERVAL_REC_LENGTH_IN_MINUTES && !decreasingInterval) {
+                } else if (newIntervalMinute >= MAX_INTERVAL_REC_LENGTH_IN_MINUTES && !decreasingInterval) {
                     IntervalTimeConfiguration givenInterval = (IntervalTimeConfiguration) anomalyDetector.getDetectionInterval();
                     this.detectorInterval = new IntervalTimeConfiguration(
-                        (long) Math.ceil(convertIntervalToMinutes(givenInterval) * INTERVAL_RECOMMENDATION_DECREASING_MULTIPLIER),
+                        (long) Math
+                            .floor(
```

Comment:
If we're still increasing, shouldn't this still be `Math.ceil()`?

Reply:
In this block it will be the first time we start decreasing. I'll change comment wording to signify that I meant this block will be executed when we haven't decreased yet but we will start to decrease within execute of the code inside this if case